File: slp/backend-dotnet/Features/Quiz/QuizQuestionDTO.cs
using System;

namespace backend_dotnet.Features.Quiz;

public class QuizQuestionDto
{
    public int Id { get; set; }
    public int QuizId { get; set; }
    public int? OriginalQuestionId { get; set; }
    public string? QuestionSnapshotJson { get; set; }
    public int DisplayOrder { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

public class CreateQuizQuestionDto
{
    public int? OriginalQuestionId { get; set; }
    public string? QuestionSnapshotJson { get; set; }
    public int DisplayOrder { get; set; }
}

public class UpdateQuizQuestionDto
{
    public int? OriginalQuestionId { get; set; }
    public string? QuestionSnapshotJson { get; set; }
    public int? DisplayOrder { get; set; }
}

File: slp/backend-nestjs/src/modules/quiz/dto/quiz-question.dto.ts
import { Expose } from 'class-transformer';
import { IsOptional, IsInt, IsString, Min } from 'class-validator';

export class CreateQuizQuestionDto {
  @IsOptional()
  @IsInt()
  @Expose()
  originalQuestionId?: number;

  @IsOptional()
  @IsString()
  @Expose()
  questionSnapshotJson?: string;

  @IsInt()
  @Min(0)
  @Expose()
  displayOrder: number;
}

export class UpdateQuizQuestionDto {
  @IsOptional()
  @IsInt()
  @Expose()
  originalQuestionId?: number;

  @IsOptional()
  @IsString()
  @Expose()
  questionSnapshotJson?: string;

  @IsOptional()
  @IsInt()
  @Min(0)
  @Expose()
  displayOrder?: number;
}

// Cũng nên thêm @Expose() cho DTO response để tránh vấn đề tương tự
export class QuizQuestionDto {
  @Expose()
  id: number;

  @Expose()
  quizId: number;

  @Expose()
  originalQuestionId?: number | null;

  @Expose()
  questionSnapshotJson?: string | null;

  @Expose()
  displayOrder: number;

  @Expose()
  createdAt: Date;

  @Expose()
  updatedAt: Date;
}